pontoon/base/static/js/translate.js

```diff
@@ -1076,6 +1166,133 @@ var Pontoon = (function (my) {
+      $('#entitylist .entity').addClass('selected');
+
+      this.getEntities({pkOnly: true}).then(function(data) {
+        self.selectedEntities = data.entity_pks;
+        self.openBatchEditor();
+      });
+    },
+
+
+    /*
+     * Attach event handlers to batch editor elements
+     */
+    attachBatchEditorHandlers: function () {
+      var self = this;
+
+      // Clear selection
+      $("#sidebar .quit-batch-editing").click(function (e) {
+        e.stopPropagation();
+        e.preventDefault();
+
+        if (self.app.advanced) {
+          self.openFirstEntity();
+        } else {
+          self.goBackToEntityList();
+        }
+      });
+
+      // Select All
+      $("#sidebar .select-all").click(function (e) {
+        e.stopPropagation();
+        e.preventDefault();
+
+        self.selectAllEntities();
+      });
+
+      // Actions
+      $('#approve-all, #delete-all, #replace-all').click(function(e) {
+        e.stopPropagation();
+        e.preventDefault();
+
+        var button = this,
+            action = $(this).attr('id').split('-')[0],
+            entities = self.selectedEntities,
+            find = $('#batch .find').val(),
+            replace = $('#batch .replace').val(),
+            message = '',
+            match = false;
+
+        // Disable before request complete
+        if ($(button).is('.loading')) {
+          return;
+        }
+
+        // Find and Replace validation
+        if (action === 'replace') {
+          if (find === '') {
+            return $('#batch .find').focus();
+          }
+
+          if (find === replace) {
+            return $('#batch .replace').focus();
+          }
+        }
+
+        $(button).addClass('loading');
+
+        $.ajax({
+          url: '/batch-edit-translations/',
+          type: 'POST',
+          data: {
+            csrfmiddlewaretoken: $('#server').data('csrf'),
+            locale: self.locale.code,
+            action: action,
+            entities: entities,
+            find: find,
+            replace: replace
+          },
+          success: function(data) {
+            if ('count' in data) {
+              var strings = data.count === 1 ? 'string' : 'strings';
+              message = data.count + ' ' + strings + ' ' + action + 'd';
+
+              // Update UI
+              if (data.count > 0) {
+                match = true;
+                setTimeout(function() {
+                  if (!self.app.advanced) {
+                    self.goBackToEntityList();
+                  }
+                  self.initializePart(true);
+                }, 1000);
+
+              }
+
+            // Empty translations produced by replace might not be always allowed
+            } else if (data.error) {
+              message = data.error;
+              $('#batch .replace').focus();
+            }
+          },
+          error: function(error) {
+            message = 'Oops, something went wrong';
+          },
+          complete: function() {
```

Comment:
It's a philosophical thing but maybe it's better to handle success/error in the complete() ? it would be easier to track (i mean read) what happens with the message.